### PR TITLE
Use BFE & ssb-box2

### PR DESCRIPTION
**Pull request by arj03**
This PR extracts the basic helper functions into [ssb-box2](https://github.com/ssbc/ssb-box2) (not super attached to the name if you want something else). This means that ssb-tribes will now only be focused on the specific way groups are implemented including the key store. And allows us to reuse ssb-box2 in db2 for box2 encrypted bendy butt messages. 

Because ssb-box2 uses BFE this PR also changes the library to use ssb-bfe for consistency.

I'm leaving this as a draft PR because this depends on https://github.com/ssbc/envelope-js/pull/10 (same rules with github things in package.json applies here). I hope that by including this PR, it will make it easier to review the envelope-js PR.

### CODE REVIEW

**Comment by arj03** [comment]
@mixmix

**Comment by arj03** [comment]
Maybe one thing to add here is that I have tests running in db2 that shows it working against a peer running db1 & tribes.

File: envelope.js [hunk 1]
```diff
@@ -78,8 +78,8 @@ module.exports = function Envelope (keystore, state) {
     // TODO change unboxer signature to allow us to optionally pass variables
     // from key() down here to save computation
     const envelope = Buffer.from(ciphertext.replace('.box2', ''), 'base64')
-    const feed_id = new FeedId(author).toTFK()
-    const prev_msg_id = new MsgId(previous).toTFK()
+    const feed_id = bfe.encode(author)
+    const prev_msg_id = bfe.encode(previous)
```
Comment (mixmix) [inline envelope.js:82]: this is a really lovely simplification. Like this API
now I understand why you had to explicitly add null support

File: envelope.js [hunk 2]
```diff
@@ -54,8 +54,8 @@ module.exports = function Envelope (keystore, state) {
     if (!isEnvelope(ciphertext)) return null
 
     const envelope = Buffer.from(ciphertext.replace('.box2', ''), 'base64')
-    const feed_id = new FeedId(author).toTFK()
-    const prev_msg_id = new MsgId(previous).toTFK()
+    const feed_id = bfe.encode(author)
+    const prev_msg_id = bfe.encode(previous)
```
Comment (mixmix) [inline envelope.js:58]: :fire: do we need to check if this this is being handed `null` ?

File: envelope.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ module.exports = function Envelope (keystore, state) {
     const plaintext = Buffer.from(JSON.stringify(content), 'utf8')
     const msgKey = new SecretKey().toBuffer()
 
-    const previousMessageId = new MsgId(previousFeedState.id).toTFK()
+    const previousMessageId = bfe.encode(previousFeedState.id)
```
Comment (mixmix) [inline envelope.js:46]: :fire: do we need to check if this this is being handed `null` ?

File: method/group/init.js [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-const Secret = require('../../lib/secret-key')
+const { SecretKey } = require('ssb-box2')
 const groupId = require('../../lib/group-id')
 const { isValid } = require('../../spec/group/init')
+const bfe = require('ssb-bfe')
 
 module.exports = function GroupCreate (ssb, keystore, state) {
   return function groupCreate (cb) {
-    const groupKey = new Secret()
+    const groupKey = new SecretKey()
```
Comment (mixmix) [inline method/group/init.js:11]: nice rename

File: test/helpers/index.js [hunk 5]
```diff
@@ -10,10 +10,10 @@ const print = require('./print')
...
 
 module.exports = {
-  GroupId: () => `%${new Secret().toString()}.cloaked`,
-  GroupKey: () => new Secret().toBuffer(),
-  FeedId: () => new FeedId().mock().toTFK(),
-  PrevMsgId: () => new MsgId().mock().toTFK(),
+  GroupId: () => `%${new SecretKey().toString()}.cloaked`,
+  GroupKey: () => new SecretKey().toBuffer(),
+  FeedId: () => new FeedId().mock().toSSB(),
+  MsgId: () => new MsgId().mock().toSSB(),
```
Comment (mixmix) [inline test/helpers/index.js:16]: I like this change - everything that is exported is now in the same format (ssb strings)

File: test/lib/cipherlinks.test.js [hunk 6]
```diff
@@ -1,38 +1,13 @@
...
-
-  const expected = Buffer.concat([
-    Buffer.from([0]), // type = feed
-    Buffer.from([0]), // format = ssb/classic
-    Buffer.from('YXkE3TikkY4GFMX3lzXUllRkNTbj5E+604AkaO1xbz8=', 'base64')
-  ])
-
-  t.deepEqual(id.toTFK(), expected, 'toTFK')
-  t.deepEqual(id.toSSB(), feedId, 'toSSB')
-
```
Comment (mixmix) [inline test/lib/cipherlinks.test.js:17]: I assume this is now all tested over in `ssb-bfe`?

**Comment by mixmix** [comment]
## Summary
@arj03 I'm loving what you've done here so far <3 It's so nice to have a reason to extract some of the complexity out of this beast.
I've moved this repo back to ssbc - I took it becaise we'd run out of free CI with travis (I think?) and I didn't have time to move to fix upgrade to gh actions. Feel free to mess with that.

looks like this refactor isn't quite ready to merge. Seems to be caught on cloning those branched repos?

File: envelope.js (hunk 2 above)
Comment (arj03) [reply envelope.js:58]: that is the responsiblity of envelope-js

File: test/lib/cipherlinks.test.js (hunk 6 above)
Comment (arj03) [reply test/lib/cipherlinks.test.js:17]: yes

**Comment by mixmix** [comment]
continued here: https://github.com/ssbc/ssb-tribes/pull/44

**Pull request closed without merging**